frontend/pages/index.tsx

```diff
@@ -13,7 +17,27 @@ import ExternalLink from "@/components/ExternalLink";
  * The component for the home page
  */
 const HomePage = () => {
+  const router = useRouter();
   const { username } = useAuthContext();
+  const [ email, setEmail ] = useState("");
+
+  const handleEmailChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    setEmail(event.target.value);
+  }
+
+  const handleSubmit = async (event: React.ChangeEvent<HTMLFormElement>) => {
+    event.preventDefault();
+    
+    const waitlistEntryData = { email };
+
+    const response = await requestWithoutAuthentication('POST', 'waitlist_entry/', waitlistEntryData);
+
+    if (response) {
+      if (response.status === 201) {
+        await router.push('/');
```

Comment:
If successful, we should clear the text field by setting `email` to `""`.

Reply:
Implemented that as well.